### PR TITLE
[common] Range-check the digit guard in DateTimeUtils

### LINKED ISSUE #9648: [Bug] A date or time component too large for an int throws instead of parsing to null
### Search before asking

- [x] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Paimon version

master, `475be566f` (2.1-SNAPSHOT).

### Compute Engine

Any. `DateTimeUtils.parseDate` and `parseTime` back the STRING to DATE and STRING to TIME casts, reached from partition value parsing, column default values, filter literals pushed down from Flink, CDC ingestion and the Hive output format.

### Minimal reproduce step

Cast a date string whose year does not fit in an `int`:

```java
DateTimeUtils.parseDate("2147483648-01-01");
// java.lang.NumberFormatException: For input string: "2147483648"
```

Every other unparseable input returns null from these methods, including an out-of-range year:

```java
DateTimeUtils.parseDate("99999-01-01");   // null, isIllegalDate rejects y > 9999
DateTimeUtils.parseDate("not-a-date");    // null
```

The guard in front of each `Integer.parseInt` only looks at the characters:

```java
private static boolean isInteger(String s) {
    boolean isInt = s.length() > 0;
    for (int i = 0; i < s.length(); i++) {
        if (s.charAt(i) < '0' || s.charAt(i) > '9') {
            isInt = false;
            break;
        }
    }
    return isInt;
}
```

So a string of 11 digits passes and `Integer.parseInt` throws. Month, day and the time components behave the same way: `parseTime("2147483648:00:00")` throws as well.

### What doesn't meet your expectations?

These methods return `Integer` and answer null for input they cannot parse; a value too large for an `int` is such an input. Instead a `NumberFormatException` escapes into the cast path, where `BinaryStringUtils.toDate` would otherwise turn the null into the `DateTimeException` that reports the bad value.

### Anything else?

Fixing the guard changes the exception a caller sees for these inputs from `NumberFormatException` to the `DateTimeException` the cast raises for any other unparseable string. One place behaves differently rather than just reporting differently: the Hive `PaimonTimeObjectInspector.convert` returns the `Integer` as is, so a TIME column with such a value writes NULL instead of failing, which is what already happens for other invalid times like `"25:00:00"`.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by LuciferYang**
### Purpose

close #9648

`DateTimeUtils.parseDate` and `parseTime` return null for input they cannot parse, and they guard every `Integer.parseInt` with `isInteger`. That guard only checked the characters:

```java
private static boolean isInteger(String s) {
    boolean isInt = s.length() > 0;
    for (int i = 0; i < s.length(); i++) {
        if (s.charAt(i) < '0' || s.charAt(i) > '9') {
            isInt = false;
            break;
        }
    }
    return isInt;
}
```

A component of eleven digits therefore passed it and `Integer.parseInt` threw `NumberFormatException`, escaping a method whose every other failure path returns null. `parseDate("2147483648-01-01")` and `parseTime("2147483648:00:00")` both do it.

The guard now also checks the range, which is enough because it sits in front of each of the eleven `parseInt` calls in those two methods. Nothing else changes: no signature, no local variable type, no arithmetic.

Callers see the difference as an exception type. Through the casts, `NumberFormatException` becomes the `DateTimeException` that `BinaryStringUtils.toDate` raises for any unparseable string, which is what a caller already gets for `"99999-01-01"` (rejected by `isIllegalDate`) or `"not-a-date"`. The Hive `PaimonTimeObjectInspector.convert` passes the `Integer` through, so a TIME column holding such a value writes NULL rather than failing, matching what it already does for other invalid times.

### Tests

`DateTimeUtilsTest.testParseDateAndTimeOverflowReturnsNull` covers a too-large year, month, day and hour, and the two boundary cases: `2147483648` is the smallest ten-digit value that does not fit an `int`, and `2147483647` does fit but is still not a valid year, so both have to come back null through different branches. Valid values are asserted alongside them.

Against the unfixed guard the test errors with `NumberFormatException: For input string: "2147483648"`.

`mvn -pl paimon-common -Dtest=DateTimeUtilsTest test` on JDK 8: 12 tests, 0 failures. `spotless:check` and `checkstyle:check` on paimon-common are clean.

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java [hunk 1]
```diff
@@ -349,15 +349,22 @@ public static Integer parseTime(String v) {
...
     }
 
+    /**
+     * Whether the string is a non-negative decimal integer that fits in an {@code int}. Callers
+     * hand the string straight to {@link Integer#parseInt}, so the range matters as much as the
+     * characters.
+     */
     private static boolean isInteger(String s) {
-        boolean isInt = s.length() > 0;
+        if (s.isEmpty() || s.length() > 10) {
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/utils/DateTimeUtils.java]: [P2] Range-check the value without rejecting leading zeros. This length limit rejects components whose numeric value still fits an int: parseDate("00000002024-01-01") and parseTime("00000000012:30:00") return 19723 and 45000000 on the base revision, but both return null with this change (confirmed in an isolated base/head probe). Through PaimonTimeObjectInspector.convert, previously accepted zero-padded TIME values are therefore silently replaced with NULL. Ignore leading zeros for the range check, or use an overflow-checked accumulator, and add regression coverage alongside the genuinely out-of-range inputs.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
The overflow guard has a real caller through Hive's TIME conversion, and the 12 DateTimeUtils tests pass with the changed class. There is a narrow compatibility regression in the new length shortcut that should be addressed without changing accepted numeric values.

**Review by JingsongLi — COMMENTED** [review]
Requirement fit: SUPPORTED. Implementation: CLEAN in the reviewed scope.

The overflow-checked accumulator addresses the earlier compatibility regression: long zero-padded components retain their accepted numeric values, while values above Integer.MAX_VALUE return the existing parse-failure sentinel. The accumulator exits before it can overflow a long. This preserves the Hive TIME conversion behavior for valid padded inputs and handles malformed overflow inputs consistently.

DateTimeUtilsTest passed all 12 cases with the changed class on JDK 8, including padded dates/times and actual overflow. No new blocking defect found; Hive integration was not executed.

**Comment by LuciferYang** [comment]
Thank you @JingsongLi
